### PR TITLE
Bs 14312 generated script edition

### PR DESCRIPTION
**Pull request by azara**

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/ContractInputGenerationWizard.java [hunk 1]
```diff
@@ -153,6 +174,21 @@ public boolean performFinish() {
         return false;
     }
 
+    /**
+     * @return
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/ContractInputGenerationWizard.java]: useless javadoc, fill it or remove it

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/ContractInputGenerationWizard.java [hunk 2]
```diff
@@ -153,6 +174,21 @@ public boolean performFinish() {
...
+     */
+    protected RootContractInputGenerator getRootContractGenerator() {
+        RootContractInputGenerator contractInputGenerator;
+        if (generatedScriptPreviewPage != null && generatedScriptPreviewPage.getRootContractInputGenerator() != null) {
+            contractInputGenerator = generatedScriptPreviewPage.getRootContractInputGenerator();
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/ContractInputGenerationWizard.java]: useless empty line

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 3]
```diff
@@ -275,12 +283,27 @@ public Object convert(final Object selectedData) {
...
     }
 
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.wizard.WizardPage#canFlipToNextPage()
+     */
+    @Override
+    public boolean canFlipToNextPage() {
+        if (generationOptions.isAutogeneratedScript()) {
+            return true;
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: directly return generationOptions.isAutogeneratedScript()

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        this.expressionBuilder = expressionBuilder;
+        this.repositoryAccessor = repositoryAccessor;
+        this.sourceViewerFactory = sourceViewerFactory;
+        generatedExpression = ExpressionFactory.eINSTANCE.createExpression();
+    }
+
+    public void setDescription() {
+        if (selectedDataObservable.getValue() != null) {
+            setDescription(Messages.bind(Messages.setGeneratedScriptPreviewPageDescription, ((Element) selectedDataObservable.getValue()).getName()));
+            EMFObservables.observeDetailValue(Realm.getDefault(), selectedDataObservable, ProcessPackage.Literals.ELEMENT__NAME).addValueChangeListener(
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java]: seems strange that we need to retrieve the default Realm

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.dialogs.IDialogPage#createControl(org.eclipse.swt.widgets.Composite)
+     */
+    @Override
+    public void createControl(final Composite parent) {
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java]: useless empty line

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        createScriptNameComposite(mainComposite);
+        createScriptPreviewComposite(mainComposite);
+        setControl(mainComposite);
+    }
+
+    /**
+     * @param mainComposite
+     */
+    protected void createScriptNameComposite(final Composite mainComposite) {
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java]: useless empty line

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java (hunk 4 above)
Comment (azara) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java]: It is same code as the method setTitle of CreateContractInputFromBusinessObjectWizardPage

**Comment by apupier** [comment]
- "Suppr" key is working in Script preview
- Script became editable after clicking on "Suppr" key and then Ctrl-z is not working

**Comment by apupier** [comment]
I would expect to have the script grayed out; if it is a preview. Perhaps a next iteration, perhaps it will become unreadable, to thought

**Comment by apupier** [comment]
if you unselect the root element, then recheck the generated scripts doesn't include the attributes.

**Comment by apupier** [comment]
Adding a unit test on FieldToContractInputMappingViewerCheckStateManager for the last behavior fixed seems a good idea to me

File: bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java [hunk 7]
```diff
@@ -104,6 +109,9 @@ protected CreateContractInputFromBusinessObjectWizardPage(final Contract contrac
         this.contract = contract;
         this.generationOptions = generationOptions;
         this.businessObjectStore = businessObjectStore;
+        this.rootNameObservable = rootNameObservable;
+        this.fieldToContractInputMappingsObservable = fieldToContractInputMappingsObservable;
+
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java]: useless extra line

**Pull request merged**